android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -369,7 +369,11 @@ public void onReceiveValue(String value) {
     }
 
     public void onMessage(String message) {
-      dispatchEvent(this, new TopMessageEvent(this.getId(), message));
+      if (mRNCWebViewClient != null) {
+        WritableMap eventData = mRNCWebViewClient.createWebViewEvent(this, this.getUrl());
+        eventData.putString("data", message);
+        dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
```

Comment:
@Titozzz Can you please check these changes?
```js
public void onMessage(String message) {
  if (mRNCWebViewClient != null) {
    WebView wv = this;
    wv.post(new Runnable() {
      @Override
      public void run() {
        WritableMap data = mRNCWebViewClient.createWebViewEvent(wv, wv.getUrl());
        data.putString("data", message);
        dispatchEvent(wv, new TopMessageEvent(wv.getId(), data));
      }
    });
  } else {
    WritableMap eventData = Arguments.createMap();
    eventData.putString("data", message);
    dispatchEvent(this, new TopMessageEvent(this.getId(), eventData));
  }
}
```
Currently your code throws
```
java.lang.Throwable: A WebView method was called on thread 'JavaBridge'. All WebView methods must be called on the same thread. (Expected Looper Looper (main, tid 2) {eead3fd} called on Looper (JavaBridge, tid 1541) {63b1022}, FYI main Looper is Looper (main, tid 2) {eead3fd})
```
exception inside `this.getUrl()`.

Also i've noticed that my changes return empty `data` in onMessage handler.

To resolve the conflict you can use this code for `TopMessageEvent.kt`:
```
package com.reactnativecommunity.webview.events

import com.facebook.react.bridge.WritableMap
import com.facebook.react.uimanager.events.Event
import com.facebook.react.uimanager.events.RCTEventEmitter

/**
 * Event emitted when there is an error in loading.
 */
class TopMessageEvent(viewId: Int, private val mEventData: WritableMap) :
    Event<TopMessageEvent>(viewId) {
    companion object {
        const val EVENT_NAME = "topMessage"
    }

    override fun getEventName(): String = EVENT_NAME

    override fun canCoalesce(): Boolean = false

    override fun getCoalescingKey(): Short = 0

    override fun dispatch(rctEventEmitter: RCTEventEmitter) =
        rctEventEmitter.receiveEvent(viewTag, eventName, mEventData)
}
```